#13 - Implementar rota que permita a listagem de usuários administrativos

atenokida opened the pull request:
## Descrição
Implementado rota para listar usuários administrativos.


## Contexto
**motivação:** 
- Implementar rota que possibilite consultar todos os usuários ativos que possuem cargo ``admin``.
- Novo contrato no _port de users_ que possibilita buscar usuários por filtros.

closes #13

<!-- Providência qualquer detalhe extra que achar importante -->

## Como testar
Enviar requisição ``GET`` para o caminho: ``/users/admin``
<!-- Explique o passo a passo de como testar essa Issue -->

### Escritor
- [x] A rota foi mapeada corretamente.
- [x] Caso ocorra alguma falha inesperada durante a busca, lança a exceção 500.
- [x] Caso não ocorra exceções, retorna as informações dos "Usuários Admins" disponíveis com código 200.

### Tester
- [x] A rota foi mapeada corretamente.
- [x] Caso ocorra alguma falha inesperada durante a busca, lança a exceção 500.
- [x] Caso não ocorra exceções, retorna as informações dos "Usuários Admins" disponíveis com código 200.

## Natureza da alteração
<!-- Indica o tipo de alteração que foi feita -->
- [X] Nova Feature.
- [ ] Correção de BUG.
- [ ] Refatoração de Código.

atenokida committed:
style: Fix spacing on `CreateUserInput`

atenokida committed:
feat: Add new port for user repository

atenokida committed:
Update CHANGELOG.md

Added task #13

atenokida committed:
feat(users): Implement `FindAll()` adapter

By default, deleted users doesn't returns

atenokida committed:
feat(users): Implement build many entities from model

atenokida committed:
fix(users): Update return type on ``FindAll()`` port

atenokida committed:
feat(users): Implement ``GetUsers()`` on service

By default: always search for users with `admin` role.

atenokida committed:
feat(users): Implement ``GetAdmins()`` on controller

atenokida committed:
feat(users): Add ``get`` route to retrieve admins

atenokida committed:
style(users): Fix typo

atenokida committed:
Revert "style(users): Fix typo"

This reverts commit cf65e475e0ab061a29e43e8a9dbcf3040ecb1c8b.

afmireski commented on internal/services/users-service.go:
```diff
@@ -180,6 +180,22 @@ func (s *UsersService) GetUserById(id string) (*entities.User, *customErrors.Int
 }
 
+func (s *UsersService) GetUsers(where myTypes.Where) ([]entities.User, *customErrors.InternalError) {
+	if len(where) == 0 {
+		where = myTypes.Where{
+			"role": map[string]string{"eq": "admin"},
+		}
+	}
+
+	repositoryData, err := s.repository.FindAll(where)
```
Aqui poderia só passar o `where`, não precisaria verificar o tamanho e passar por padrão `role admin`. No seu método no controller fica mais certo fazer isso.
```suggestion
	repositoryData, err := s.repository.FindAll(where)
```

atenokida replied:
Beleza. 
Só para fins de documentação: tinha feito dessa forma pois na especificação estava "Na lógica do service por padrão só serão listados adms".

afmireski commented on internal/web/controllers/users-controller.go:
```diff
@@ -114,6 +114,21 @@ func (c *UsersController) GetUserById(w http.ResponseWriter, r *http.Request) {
+func (c *UsersController) GetAdmins(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	users, serviceErr := c.service.GetUsers(nil)
```
Seguindo o comentário do `service`:
```suggestion
	users, serviceErr := c.service.GetUsers(myTypes.Where{
		"role": map[string]string{"eq": "admin"},
	})
```

afmireski reviewed (changes requested):
Só corrigir um pequeno detalhe.

atenokida committed:
Update internal/web/controllers/users-controller.go

Co-authored-by: André Felipe Mireski <54330175+afmireski@users.noreply.github.com>

atenokida committed:
Update internal/services/users-service.go

Co-authored-by: André Felipe Mireski <54330175+afmireski@users.noreply.github.com>

afmireski reviewed (approved):
Aparentemente tudo certo.

afmireski committed:
#13 - refactor: Fix some little serialization problem

- If the list of adms was empty, null was returned. Change to '[]'
- Remove unecessary LOGS
